### PR TITLE
Pract.02: Add 01-03 | Ivan Angelov

### PR DESCRIPTION
**Pull request by ivan-angelov9643**
Ivan Angelov 
SI 5 9MI0600190

### CODE REVIEW

File: Seminars/Sem.02/Pract.02/Task01.cpp [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+void PrintFileContent(const char* fileName) {
+	ifstream fileToRead(fileName);
+	if (!fileToRead.is_open())
+	{
+		cout << "Couldn't open " << fileName << endl;
+		return;
+	}
+	while (!fileToRead.eof())
+	{
+		char buffer[128];
```
Comment (GeorgiTerziev02) [inline Seminars/Sem.02/Pract.02/Task01.cpp:33]: use constant

File: Seminars/Sem.02/Pract.02/Task01.cpp [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+		fileToRead.getline(buffer, 128);
+		cout << buffer << endl;
```
Comment (GeorgiTerziev02) [inline Seminars/Sem.02/Pract.02/Task01.cpp:35]: Please check again what is required in the task

File: Seminars/Sem.02/Pract.02/Task02.cpp [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+	}
+	int result = 0;
+	while (!fileToRead.eof())
+	{
+		char buffer[1024];
+		fileToRead.getline(buffer, 1024);
+		result += CharArrayLength(buffer);
+	}
+	return result;
+}
```
Comment (GeorgiTerziev02) [inline Seminars/Sem.02/Pract.02/Task02.cpp:36]: use tellg, seekg

File: Seminars/Sem.02/Pract.02/Task03.cpp [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+		delete[] values[i];
+		delete[] placeholders[i];
+	}
+	for (size_t i = 0; i < templateLinesCount; i++)
+	{
+		delete[] templateLines[i];
+	}
+}
+
+void PrintLines(char** lines, size_t linesCount) {
```
Comment (GeorgiTerziev02) [inline Seminars/Sem.02/Pract.02/Task03.cpp:42]: const

File: Seminars/Sem.02/Pract.02/Task03.cpp [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+				{
+					templateLines[i][ind] = values[placeholderIndex][k++];
+				}
+			}
+			j++;
+		}
+	}
+}
+
+void CharArrayCopy(char* toCopy, size_t length,char* copyTo) {
```
Comment (GeorgiTerziev02) [inline Seminars/Sem.02/Pract.02/Task03.cpp:85]: const

File: Seminars/Sem.02/Pract.02/template.txt [hunk 6]
```diff
@@ -0,0 +1,6 @@
+Dear, {title} {recipient_name}.
```
Comment (GeorgiTerziev02) [inline Seminars/Sem.02/Pract.02/template.txt:1]: do not upload txt

**Review by GeorgiTerziev02 — CHANGES_REQUESTED** [review]

**Comment by Justsvetoslavov** [comment]
Check uploaded solutions.
You can re-open the PR or open a new one if you have better solution or questions.

**Pull request closed without merging**